### PR TITLE
[ROCm] Add script to run multi gpu tests

### PR DESCRIPTION
**Pull request by hsharsha**

### CODE REVIEW

File: build_tools/rocm/run_xla_multi_gpu.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    --keep_going \
+    --local_test_jobs=${N_TEST_JOBS} \
+    --test_env=TF_TESTS_PER_GPU=$TF_TESTS_PER_GPU \
+    --test_env=TF_GPU_COUNT=$TF_GPU_COUNT \
+    --action_env=XLA_FLAGS=--xla_gpu_force_compilation_parallelism=16 \
+    --action_env=XLA_FLAGS=--xla_gpu_enable_llvm_module_compilation_parallelism=true \
+    -- //xla/tests:collective_ops_test_e2e_gpu \
+       //xla/tests:collective_ops_test_gpu \
+       //xla/tests:replicated_io_feed_test_gpu \
+       //xla/tools/multihost_hlo_runner:functional_hlo_runner_test_gpu
```
Comment (i-chaochen) [inline build_tools/rocm/run_xla_multi_gpu.sh:72]: Thanks! We need to add these two tests as well.

```
//xla/pjrt/distributed:topology_util_test
//xla/pjrt/distributed:client_server_test
```

File: build_tools/rocm/run_xla_multi_gpu.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+N_BUILD_JOBS=$(grep -c ^processor /proc/cpuinfo)
+# If rocm-smi exists locally (it should) use it to find
+# out how many GPUs we have to test with.
+rocm-smi -i
+STATUS=$?
+if [ $STATUS -ne 0 ]; then TF_GPU_COUNT=4; else
+   TF_GPU_COUNT=$(rocm-smi -i|grep 'Device ID' |grep 'GPU' |wc -l)
+fi
+TF_TESTS_PER_GPU=1
+N_TEST_JOBS=$(expr ${TF_GPU_COUNT} \* ${TF_TESTS_PER_GPU})
```
Comment (i-chaochen) [inline build_tools/rocm/run_xla_multi_gpu.sh:30]: we need to add gpu number check as well, the multiple gpu test needs at least 4 gpu.

**Comment by hsharsha** [comment]
Updated with pjrt distrbuted tests and check for number of gpus

File: build_tools/rocm/run_xla_multi_gpu.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+N_BUILD_JOBS=$(grep -c ^processor /proc/cpuinfo)
+# If rocm-smi exists locally (it should) use it to find
+# out how many GPUs we have to test with.
+rocm-smi -i
+STATUS=$?
+if [ $STATUS -ne 0 ]; then TF_GPU_COUNT=1; else
+   TF_GPU_COUNT=$(rocm-smi -i|grep 'Device ID' |grep 'GPU' |wc -l)
+fi
+if [[ $TF_GPU_COUNT -lt 4 ]]; then
+    echo "Found only ${TF_GPU_COUNT} gpus, multi-gpu tests need atleast 4 gpus."
```
Comment (i-chaochen) [inline build_tools/rocm/run_xla_multi_gpu.sh:30]: a space between `atleast` ? :)

**Review by i-chaochen — APPROVED** [review]

**Pull request merged**